### PR TITLE
feat(kit): add full-width numbers support for `Time`, `Date`, `DateTime`, `DateRange`

### LINKED ISSUE #884: 🚀 - Add full-width numbers support for `Time`, `Date`, `DateTime`, `DateRange`
### Which package(s) are relevant/related to the feature request?

@maskito/kit

### Description

1. Move [`toHalfWidthNumber`](https://github.com/taiga-family/maskito/blob/main/projects/kit/src/lib/masks/number/utils/to-half-width-number.ts) to [`projects/kit/src/lib/utils`](https://github.com/taiga-family/maskito/tree/main/projects/kit/src/lib/utils).

2. Move [`createFullWidthToHalfWidthPreprocessor`](https://github.com/taiga-family/maskito/blob/main/projects/kit/src/lib/masks/number/processors/fullwidth-to-halfwidth-preprocessor.ts) to [`projects/kit/src/lib/processors`](https://github.com/taiga-family/maskito/tree/main/projects/kit/src/lib/processors).

3. And add `createFullWidthToHalfWidthPreprocessor` to  [`Time`](https://maskito.dev/kit/time), [`Date`](https://maskito.dev/kit/date), [`DateRange`](https://maskito.dev/kit/date-range), [`DateTime`](https://maskito.dev/kit/date-time).

4. Add Cypress Tests for full-width cases

Learn more:
* https://github.com/taiga-family/maskito/pull/864#discussion_r1447423820

### PR DESCRIPTION
**Pull request by rikusen0335**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #884

This PR implements full-width to half-width preprocessor into time, or datetime or almost similar masks.

Please see issue page for more detail

## What is the new behaviour?

Above masks will have full-width to half-width preprocessor.

## Implementation
- Moved `createFullWidthToHalfWidthPreprocessor` from `projects/kit/src/lib/masks/number/processors` to `projects/kit/src/lib/processors` (also tests)
- Added `createFullWidthToHalfWidthPreprocessor` to `Time`, `Date`, `DateTime`, `DateRange` masks
- Added e2e tests for `Time`, `Date`, `DateTime`, `DateRange` masks

### CODE REVIEW

**Comment by rikusen0335** [comment]
@nsbarsukov
Made a tests for Date or Time related, please review this when you have a time :)

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/kit/src/lib/masks/date-range/date-range-mask.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
 } from '../../processors';
 import {MaskitoDateMode, MaskitoDateSegments} from '../../types';
 import {parseDateRangeString} from '../../utils';
+import {createFullWidthToHalfWidthPreprocessor} from '../number/processors';
```
[the remark this answers is not in the document]
Comment (rikusen0335) [reply projects/kit/src/lib/masks/date-range/date-range-mask.ts:14]: Oh, sorry it accidently revert the changes. Will work on this

**Comment by rikusen0335** [comment]
Okay, now I added colon convert preprocessor to `Time` and `DateTime` masks.

**Review by nsbarsukov — APPROVED** [review]

**Comment by nsbarsukov** [comment]
@rikusen0335 Thanks ❤️

**Comment by rikusen0335** [comment]
Congrats! 🎉

**Pull request merged**